File: lesson_11/arrays_java/arrays_app/src/main/java/com/codedifferently/lesson11/Lesson11.java
package com.codedifferently.lesson11;

import java.util.ArrayList;
import java.util.List;

public class Lesson11 {

  /**
   * Provide the solution to LeetCode 1929 here:
   * https://leetcode.com/problems/concatenation-of-array
   */
  public int[] getConcatenation(int[] nums) {
    int ans[] = new int[2 * nums.length];

    System.arraycopy(nums, 0, ans, 0, nums.length);
    System.arraycopy(nums, 0, ans, nums.length, nums.length);
    return ans;
  }

  // worked on the solutions with Pablo, Joseph, & Nile via jitsi. //
  /**
   * Provide the solution to LeetCode 2942 here:
   * https://leetcode.com/problems/find-words-containing-character/
   */
  public List<Integer> findWordsContaining(String[] words, char x) {
    List<Integer> indices = new ArrayList<>();
    for (int i = 0; i < words.length; i++) {
      String str = String.valueOf(x);
      if (words[i].contains(str)) {
        indices.add(i);
      }
    }
    return indices;
  }
}

File: lesson_11/arrays_java/arrays_app/src/main/java/com/codedifferently/lesson11/Lesson11.ts
/**
* Provide the solution to LeetCode 1929 here:
* https://leetcode.com/problems/concatenation-of-array
*/

function getConcatenation(nums: number[]): number[] {
    let n: number = nums.length;
    let ans: number[] = new Array(2*n);
    
     for(let i = 0; i < n; i++){
        ans[i] = nums[i];
        ans[i + n] = nums[i];
       }

    return ans;
};